lumberjack.go

```diff
@@ -324,9 +339,30 @@ func deleteAll(dir string, files []logInfo) {
 	}
 }
 
+// compressLogs compresses any uncompressed logs during the cleanup process
+func (l *Logger) compressLogs() {
+	l.cmu.Lock()
+	defer l.cmu.Unlock()
+	files, err := l.oldLogFiles(false)
+	if err != nil {
+		fmt.Errorf("Unable to compress log files: %s", err)
+	}
+
+	for _, file := range files {
+		_, ext := l.prefixAndExt()
+		if ext != compressFileExtension {
+			if err := compressLog(filepath.Join(l.dir(), file.Name())); err != nil {
+				fmt.Errorf("Unable to compress backup log file: %s", err)
+			}
+		}
+	}
+}
+
 // oldLogFiles returns the list of backup log files stored in the same
-// directory as the current log file, sorted by ModTime
-func (l *Logger) oldLogFiles() ([]logInfo, error) {
+// directory as the current log file, sorted by ModTime. Setting
+// includeCompressed to true will include files with the given
+// compressFileExtension into the returned list
+func (l *Logger) oldLogFiles(includeCompressed bool) ([]logInfo, error) {
```

Comment:
includeCompressed is not doing what it seems like you're saying it's doing.  It's not one or both, it's one or the other.  When you use includeCompressed, it'll never return uncompressed files, because they don't have a matching extension.  That's not exactly ok, since this could skip a file that hasn't been compressed yet.